test/type/index.ts

```diff
@@ -11,3 +11,11 @@ getCodecFromSerialized<A>({ A1: t.string, A2: t.number }) // $ExpectError
 getCodecFromSerialized<A>({ A1: t.undefined, A2: t.number }) // $ExpectError
 getCodecFromSerialized<A>({ A2: t.number }) // $ExpectError
 getCodecFromSerialized<A>({ A1: t.null }) // $ExpectError
+
+type B = Sum.Member<"B1"> | Sum.Member<"B2">
+
+getCodecFromNullaryTag<B>()([]) // $ExpectError
+getCodecFromNullaryTag<B>()(["B1"]) // $ExpectError
+getCodecFromNullaryTag<B>()(["B2"]) // $ExpectError
+getCodecFromNullaryTag<B>()(["B1", "B1"]) // $ExpectError
+getCodecFromNullaryTag<B>()(["B1", "B2"]) // $ExpectType Type<B, string, unknown>
```

Comment:
What happens if we try to use this with a sum type that contains a non-nullary member?

Reply:
Totally forgot about this PR haha.

This'll happen on the generic:

```
Diagnostics:
1. Type 'B' does not satisfy the constraint 'Member<string, null>'.
     Type 'Member<"B1", number>' is not assignable to type 'Member<string, null>'.
       Type 'number' is not assignable to type 'null'.
```